### PR TITLE
All algorithms

### PR DESCRIPTION
**Pull request by alan-stokes**
tie in for the project to test mapping algorithms. is a blind copy of c model with bitfields

### CODE REVIEW

File: spinn_utilities/executable_finder.py [hunk 1]
```diff
@@ -56,20 +58,22 @@ def get_executable_path(self, executable_name):
                 return potential_filename
 
         # No executable found
-        return None
+        raise FailedToFindBinaryException(
```
Comment (dkfellows) [inline spinn_utilities/executable_finder.py]: Why are you introducing a new failure mode?

Comment (alan-stokes) [reply spinn_utilities/executable_finder.py]: better to fail before loading data. if you dont have a failure mode here. it causes some horrible error about flood filling None. fail fast and understandablly

**Comment by alan-stokes** [comment]
fixed elsewhere

**Comment by alan-stokes** [comment]
alas not. we do need this. at least for a file rename

**Pull request merged**